keras/backend/tensorflow_backend.py

```diff
@@ -1202,8 +1202,11 @@ def rnn(step_function, inputs, initial_states,
     if mask is not None:
         if mask.dtype != tf.bool:
             mask = tf.cast(mask, tf.bool)
-        if len(mask.get_shape()) == ndim - 1:
-            mask = expand_dims(mask)
+        if len(mask.get_shape()) < ndim:
+            tmp = ndim - len(mask.get_shape())
+            for _ in range(tmp):
```

Comment:
Can get rid of `tmp` by putting the expression inside `range`. Would be cleaner. Else, rename `tmp` to something explicit.